### DISCUSSION TITLE
MST over surface of sphere?

### DISCUSSION THREAD

**Discussion by TadeasManka** [NanoComp/meep#2034] [Q&A] [answered]
Hello everyone,
I am calculating the optical force acting on dielectric sphere.
Is there a way to compute Maxwell-Stress Tensor on surface over sphere...instead of integrating it on the box surrounding the sphere as is in the tutorial?
Thanks for response.

**Comment by stevengj** [comment]
The integral over the sphere and the integral over an enclosing box should give the same result (for an object surrounded by vacuum, as long as there are no current sources between the box and the sphere).

**Reply by TadeasManka** [reply]
Yes but in some geometries (dielectric sphere inside cylindrical hole) it would be useful, because we cannot go closer to the edge of the hole than the hypotenuse of the side of the box

**Comment by stevengj** [comment, marked as answer]
Fair enough.   More generally you can define a sequence of planar [`ForceRegion` objects](https://meep.readthedocs.io/en/latest/Python_User_Interface/#ForceRegion) that together enclose the sphere — with enough of them, you can enclose any curved surface.

**Reply by TadeasManka** [reply]
Thats a good point, thank you!
